libraries/AC_Avoidance/AP_OADijkstra.cpp

```diff
@@ -133,27 +148,39 @@ AP_OADijkstra::AP_OADijkstra_State AP_OADijkstra::update(const Location &current
         }
     }
 
-    // rebuild path if destination has changed
-    if (!destination.same_latlon_as(_destination_prev)) {
+    // rebuild path if destination or next_destination has changed
+    if (!destination.same_latlon_as(_destination_prev) || !next_destination.same_latlon_as(_next_destination_prev)) {
         _destination_prev = destination;
+        _next_destination_prev = next_destination;
         _shortest_path_ok = false;
     }
 
     // calculate shortest path from current_loc to destination
     if (!_shortest_path_ok) {
         _shortest_path_ok = calc_shortest_path(current_loc, destination, error_id);
         if (!_shortest_path_ok) {
+            dest_to_next_dest_clear = _dest_to_next_dest_clear = false;
```

Comment:
Did you intend to do two assignments on the same line?

Reply:
Hi @Ryanf55, thanks for the review.  They're actually slightly different in that one has an underscore at the front..

Reply:
Ah, ok, just checking.